### PR TITLE
tso: replace priorityCheck `ResetLeader` with `TransferPrimary`

### LINKED ISSUE #8533: TestGetTSOImmediately is flaky
### Which jobs are failing
```
2024-08-15T06:17:50.3934438Z     testutil.go:67: 
2024-08-15T06:17:50.3935833Z         	Error Trace:	/home/runner/work/pd/pd/pkg/utils/testutil/testutil.go:67
2024-08-15T06:17:50.3938318Z         	            				/home/runner/work/pd/pd/tests/integrations/mcs/tso/keyspace_group_manager_test.go:772
2024-08-15T06:17:50.3939789Z         	Error:      	Condition never satisfied
2024-08-15T06:17:50.3940870Z         	Test:       	TestGetTSOImmediately
```

### CI link
https://github.com/tikv/pd/actions/runs/10399520880/job/28798931505

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

The issue detail process is https://github.com/tikv/pd/issues/8533#issuecomment-2331001152

tso's priority uses `ResetLeader` instead of `moveLeader`.
The process is: 
- secondary: primary check priority -> resetLeader -> secondary exit watch -> secondary enter campaign -> secondary select new leader 
- primary: 	  primary check priority -> resetLeader -> (need 200ms tick)[time gap]  primary finds out there is not primary -> exit campaign -> primary enter campaign

Secondary can be elected as new primary because of **time gap** which is not stable!!!
> For example, if the secondary io jitters and doesn't elected as new primary, the old primary will be elected, and then it will loop through the priority check logic again.

Issue Number: Close #8533

### What is changed and how does it work?

- Because [`transfer primary`](https://github.com/tikv/pd/pull/8157) is actually as `move etcd leader`.
- So replace `ResetLeader` with `transfer primary`, which is more time efficient and can avoid time lag.

```commit-message
```

### Check List

Tests

- Integration test
- Manual test (add detailed scripts or steps below)

### Release note

```release-note
tso: replace priorityCheck ResetLeader with TransferPrimary
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: pkg/tso/keyspace_group_manager.go [hunk 1]
```diff
@@ -606,7 +607,7 @@ func (kgm *KeyspaceGroupManager) primaryPriorityCheckLoop() {
 	defer kgm.wg.Done()
 
 	failpoint.Inject("fastPrimaryPriorityCheck", func() {
-		kgm.primaryPriorityCheckInterval = 200 * time.Millisecond
+		kgm.primaryPriorityCheckInterval = 1 * time.Second
```
Comment (rleungx) [inline pkg/tso/keyspace_group_manager.go:610]: Why enlarge it?

Comment (HuSharp) [reply pkg/tso/keyspace_group_manager.go:610]: I feel no need to keep calling `TransferPrimary` every 200ms, which probably won't happen in a real scenario?

Comment (rleungx) [reply pkg/tso/keyspace_group_manager.go:610]: Is it used to accelerate the speed of the test?

Comment (HuSharp) [reply pkg/tso/keyspace_group_manager.go]: got it, and have reverted it :)

**Review by rleungx — APPROVED** [review]

**Pull request merged**